back/engines/free/user_confirmation/app/models/user_confirmation/extensions/user.rb

```diff
@@ -8,24 +8,59 @@ def self.included(base)
             validates :email_confirmation_retry_count, numericality: { less_than_or_equal_to: ENV.fetch('EMAIL_CONFIRMATION_MAX_RETRIES', 5) }
             validates :email_confirmation_code_reset_count, numericality: { less_than_or_equal_to: ENV.fetch('EMAIL_CONFIRMATION_MAX_RETRIES', 5) }
 
-            before_validation :reset_confirmation_code, unless: :email_confirmation_code, if: :email_changed?
+            with_options if: :email_changed?, on: :create do
+              before_validation :reset_confirmation_code
+              before_validation :reset_confirmed_at
+              before_validation :reset_confirmation_required
+            end
+
+            before_validation :confirm, if: ->(user) { user.invite_status_change&.last == 'accepted' }
           end
         end
       end
 
-      def requires_confirmation?
-        AppConfiguration.instance.feature_activated?('user_confirmation') &&
-          !(active? || invited? || confirmed? || registered_with_phone?)
+      public
+
+      #
+      # <Used to check upon update or create, if a user should have to confirm their account>
+      #
+      # @return [<Boolean>] <True if the user requires confirmation>
+      #
+      def should_require_confirmation?
+        !(registered_with_phone? || highest_role != :user)
```

Comment:
It's just a thought, but maybe it would be nice to have a method to compare roles (higher or lower than). Suppose that a new role gets introduced that is equivalent to a user (a visitor or citizen or outsider, I don't know 😅), then a bug might get introduced where they don't need to confirm their email.

Reply:
I agree 👍